flecsi-sp/burton/burton_types.h

```diff
@@ -30,6 +30,265 @@ namespace burton {
 template< std::size_t N, bool Include_Extras = false >
 struct burton_types_t {};
 
+////////////////////////////////////////////////////////////////////////////////
+//! \brief A collection of type information needed to specialize the flecsi
+//!   low-level mesh infrastructure for ALE methods.
+//! \remark This is the one-dimensional version.
+////////////////////////////////////////////////////////////////////////////////
+struct burton_1d_types_base
+{
+
+  //============================================================================
+  // Define local traits to satisfy mesh_topology requirements.
+  //============================================================================
+
+  //! the mesh traites
+  using config_t = burton_config_t<1>;
+
+  //! The dimension of the burton mesh picked up from burton_config_t.
+  static constexpr size_t num_dimensions = config_t::num_dimensions;
+
+  //! The number of domains in burton mesh picked up from burton_config_t.
+  static constexpr size_t num_domains = config_t::num_domains;
+
+  //! the flecsi mesh topology type
+  using mesh_topology_base_t = typename config_t::mesh_topology_base_t;
+
+  //! the base type for the entities
+  using mesh_entity_base_t = typename config_t::mesh_entity_base_t;
+
+  //! the id type
+  using id_t = flecsi::utils::id_t;
+
+  //============================================================================
+  //! \brief The burton mesh index spaces.
+  //============================================================================
+  struct index_spaces_t {
+
+    //! The individual enumeration of the index spaces
+    enum index_spaces : size_t {
+      // the main index spaces
+      vertices,
+      edges = vertices,
+      faces = edges,
+      cells,
+      corners,
+      wedges = corners,
+      // index spaces for connectivity
+      vertices_to_cells,
+      edges_to_cells = vertices_to_cells,
+      faces_to_cells = vertices_to_cells,
+      cells_to_vertices,
+      cells_to_edges = cells_to_vertices,
+      cells_to_faces = cells_to_vertices,
+      // corners
+      cells_to_corners,
+      vertices_to_corners,
+      edges_to_corners = vertices_to_corners,
+      faces_to_corners = vertices_to_corners,
+      corners_to_cells,
+      corners_to_vertices,
+      corners_to_edges = corners_to_vertices,
+      corners_to_faces = corners_to_vertices,
+      // wedges
+      cells_to_wedges = cells_to_corners,
+      vertices_to_wedges = vertices_to_corners,
+      edges_to_wedges = vertices_to_corners,
+      faces_to_wedges = vertices_to_corners,
+      wedges_to_cells = corners_to_cells,
+      wedges_to_vertices = corners_to_vertices,
+      wedges_to_edges = corners_to_vertices,
+      wedges_to_faces = corners_to_vertices,
+      // index spaces that are not used
+      vertices_to_edges = 7777,
```

Comment:
What, no `Fnul`? :-)

Reply:
lol this is on me.  sorry I had to put something other than -1

Reply:
@cmsquared : Didn't you know that 7777 is the 16-bit version of `Fnul`, for use on reduced-precision GPUs and AI engines?  ;-)